// @fw-components/fw-theme-builder/models.js
export const fontOptions = [
    {
        name: "Archivo",
        url: 'https://fonts.gstatic.com/s/archivo/v18/k3kPo8UDI-1M0wlSV9XAw6lQkqWY8Q82sLydOxI.woff2',
        style: "'Archivo', sans-serif",
    },
    {
        name: "Barlow",
        url: 'https://fonts.gstatic.com/s/barlow/v12/7cHpv4kjgoGqM7E_DMs5.woff2',
        style: "'Barlow', sans-serif",
    },
    {
        name: "Cabin",
        url: 'https://fonts.gstatic.com/s/cabin/v26/u-4i0qWljRw-PfU81xCKCpdpbgZJl6Xvqdns.woff2',
        style: "'Cabin', sans-serif",
    },
    {
        name: "DM Sans",
        url: 'https://fonts.gstatic.com/s/dmsans/v11/rP2Hp2ywxg089UriCZOIHQ.woff2',
        style: "'DM Sans', sans-serif",
    },
    {
        name: "Geologica",
        url: 'https://fonts.gstatic.com/s/geologica/v1/oY1l8evIr7j9P3TN9YwNAdyjzUyDKkKdAGOJh1UlCDUIhAIdhCZOn1fLsig7jfvCCPHZckUWE1lE.woff2',
        style: "'Geologica', sans-serif",
    },
    {
        name: "Inter",
        url: 'https://fonts.gstatic.com/s/inter/v12/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa1ZL7.woff2',
        style: "'Inter', sans-serif",
    },
    {
        name: "Josefin Sans",
        url: 'https://fonts.gstatic.com/s/josefinsans/v32/Qw3aZQNVED7rKGKxtqIqX5EUAnx4Vn8siqM7.woff2',
        style: "'Josefin Sans', sans-serif",
    },
    {
        name: "Karla",
        url: 'https://fonts.gstatic.com/s/inter/v12/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2JL7SUc.woff2',
        style: "'Karla', sans-serif",
    },
    {
        name: "Nunito Sans",
        url: 'https://fonts.gstatic.com/s/nunitosans/v15/pe0TMImSLYBIv1o4X1M8ce2xCx3yop4tQpF_MeTm0lfGWVpNn64CL7U8upHZIbMV51Q42ptCp7t1R-s.woff2',
        style: "'Nunito Sans', sans-serif",
    },
    {
        name: "Playfair Display",
        url: 'https://fonts.gstatic.com/s/playfairdisplay/v30/nuFiD-vYSZviVYUb_rj3ij__anPXDTzYgA.woff2',
        style: "'Playfair Display', serif",
    },
    {
        name: "Poppins",
        url: 'https://fonts.gstatic.com/s/poppins/v20/pxiEyp8kv8JHgFVrJJfecg.woff2',
        style: "'Poppins', sans-serif",
    },
    {
        name: "Public Sans",
        url: 'https://fonts.gstatic.com/s/publicsans/v14/ijwRs572Xtc6ZYQws9YVwnNGfJ4.woff2',
        style: "'Public Sans', sans-serif",
    },
    {
        name: "Raleway",
        url: 'https://fonts.gstatic.com/s/raleway/v28/1Ptug8zYS_SKggPNyC0ITw.woff2',
        style: "'Raleway', sans-serif",
    },
    {
        name: "Roboto",
        url: 'https://fonts.gstatic.com/s/roboto/v30/KFOmCnqEu92Fr1Mu4mxK.woff2',
        style: "'Roboto', sans-serif",
    },
    {
        name: "Roboto Slab",
        url: 'https://fonts.gstatic.com/s/robotoslab/v25/BngMUXZYTXPIvIBgJJSb6ufN5qU.woff2',
        style: "'Roboto Slab', serif",
    },
    {
        name: "Rubik",
        url: 'https://fonts.gstatic.com/s/rubik/v26/iJWKBXyIfDnIV7nBrXw.woff2',
        style: "'Rubik', sans-serif",
    },
    {
        name: "Tajawal",
        url: 'https://fonts.gstatic.com/s/tajawal/v9/Iurf6YBj_oCad4k1l8KiHrFpiQ.woff2',
        style: "'Tajawal', sans-serif",
    },
];
export const defaultTheme = {
    fonts: {
        "Title": {
            name: "DM Sans",
            url: 'https://fonts.gstatic.com/s/dmsans/v11/rP2Hp2ywxg089UriCZOIHQ.woff2',
            style: "'DM Sans', sans-serif",
        },
        "Body": {
            name: "DM Sans",
            url: 'https://fonts.gstatic.com/s/dmsans/v11/rP2Hp2ywxg089UriCZOIHQ.woff2',
            style: "'DM Sans', sans-serif",
        }
    },
    sizes: {
        "Tiny": "10px",
        "XS": "14px",
        "S": "16px",
        "M": "18px",
        "L": "20px",
        "XL": "22px",
        "Huge": "48px",
    },
    colors: {
        "Primary": {
            "Hex": "#ad38d1",
            "L1": "#ba68d3",
            "L2": "#dcb5e7",
            "L3": "#e5d3eb",
            "Contrast": "#f0f0f0",
        },
        "Secondary": {
            "Hex": "#4a48c7",
            "L1": "#5f5dce",
            "L2": "#706fd1",
            "L3": "#9190df",
            "Contrast": "#ebdbdb",
        },
        "Background": {
            "Hex": "#eeeeee",
        },
        "Error": {
            "Hex": "#e61e1e",
            "L1": "#f1b4b4",
        },
        "Text": {
            "Title": "#1b1b1b",
            "Subtitle": "#1b1b1b",
            "Body": "#1b1b1b",
            "Body-l1": "#363636",
        }
    }
};
